### PR TITLE
PUB-1663 Clear Subscriptions

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
### JIRA link

https://tools.hmcts.net/jira/browse/PUB-1663

### Change description

Add page to delete the subscription for the selected court. Delete subscription link will be visible when system admin tries to delete the court but there is a subscription for that court.

image

image

image

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

File: src/main/resources/locales/en/delete-court-reference-data-confirmation.json [hunk 1]
```diff
@@ -12,5 +12,6 @@
         "locationType": "Location Type",
         "jurisdiction": "Jurisdiction",
         "region": "Region"
-    }
+    },
+    "deleteSubscriptionText": "Click here to delete the all the subscription(s) for "
```
Comment (KianKwa) [inline src/main/resources/locales/en/delete-court-reference-data-confirmation.json:16]: Think there is a redundant "the" in the sentence. Should it not be "delete all the subscription(s)..."?

File: src/main/controllers/DeleteCourtSubscriptionConfirmationController.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                req.user['email'],
+                'DELETE_LOCATION_SUBSCRIPTION_SUCCESS',
+                response.toString()
+            );
+            res.redirect('/delete-court-subscription-success');
+          }
+        } else if (formData['delete-choice'] == 'no') {
+          res.redirect('/delete-court-reference-data');
+        } else {
+        res.render('delete-court-reference-data-confirmation', {
```
Comment (KianKwa) [inline src/main/controllers/DeleteCourtSubscriptionConfirmationController.ts:40]: The line should be indented. Run the prettier command to fix this

File: src/main/resources/locales/en/delete-court-reference-data-confirmation.json (hunk 1 above)
Comment (junaidiqbalmoj) [reply src/main/resources/locales/en/delete-court-reference-data-confirmation.json]: fixed

File: src/main/controllers/DeleteCourtSubscriptionConfirmationController.ts (hunk 2 above)
Comment (junaidiqbalmoj) [reply src/main/controllers/DeleteCourtSubscriptionConfirmationController.ts]: fixed.

File: src/main/controllers/DeleteCourtSubscriptionConfirmationController.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            const response = await subscriptionsService.deleteLocationSubscription(
+                formData.locationId,
+                req.user?.['provenanceUserId']
+            );
+            if (response === null) {
+                res.render('delete-court-subscription-confirmation', {
+                    ...cloneDeep(req.i18n.getDataByLanguage(req.lng)['delete-court-subscription-confirmation']),
+                    court: locationService.formatCourtValue(court),
+                    apiError: true,
+                    errorMessage: 'Unknown error when attempting to delete all the subscription for the court',
```
Comment (ChrisS1512) [inline src/main/controllers/DeleteCourtSubscriptionConfirmationController.ts:26]: I think in the error message it should be 'subscriptions'

File: src/main/resources/locales/cy/delete-court-subscription-confirmation.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+    "header": "Are you sure you want to delete all the subscription?",
```
Comment (ChrisS1512) [inline src/main/resources/locales/cy/delete-court-subscription-confirmation.json:2]: Similar to the above, I think subscription should be subscriptions (Same below on line 5)

File: src/main/resources/locales/en/delete-court-subscription-confirmation.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+    "header": "Are you sure you want to delete all the subscription?",
```
Comment (ChrisS1512) [inline src/main/resources/locales/en/delete-court-subscription-confirmation.json:2]: Same comment as Welsh file

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

**Review by KianKwa — DISMISSED** [review]

File: src/main/controllers/DeleteCourtSubscriptionConfirmationController.ts (hunk 3 above)
Comment (junaidiqbalmoj) [reply src/main/controllers/DeleteCourtSubscriptionConfirmationController.ts]: fixed.

File: src/main/resources/locales/en/delete-court-subscription-confirmation.json (hunk 5 above)
Comment (junaidiqbalmoj) [reply src/main/resources/locales/en/delete-court-subscription-confirmation.json]: fixed.

File: src/main/resources/locales/cy/delete-court-subscription-confirmation.json (hunk 4 above)
Comment (junaidiqbalmoj) [reply src/main/resources/locales/cy/delete-court-subscription-confirmation.json]: fixed.

**Review by KianKwa — APPROVED** [review]

**Review by KianKwa — APPROVED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Comment by ChrisS1512** [comment]
Closing as covered in the 1664 PR

**Pull request closed without merging**